components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/RecoveryProcessor.java

```diff
@@ -440,6 +446,13 @@ public NotificationDataDTO notifyWithEmail(UserRecoveryDTO notificationBean) thr
         String domainName = notificationBean.getTenantDomain();
         int tenantId = notificationBean.getTenantId();
         confirmationKey = notificationBean.getConfirmationCode();
+        int index = 0;
+        String userStore = UserStoreConfigConstants.PRIMARY;
+        String userName = userId;
+        if ((index = userId.indexOf(CarbonConstants.DOMAIN_SEPARATOR)) > 0) {
```

Comment:
try to use UserCoreUtil methods like extractDomainFromName, removeDomainFromName